apps/gitness/src/pages/pipeline-edit/components/pipeline-studio-step-palette.tsx

```diff
@@ -26,112 +31,161 @@ const PipelineStudioStepPalette = (props: PipelineStudioStepFormProps): JSX.Elem
   } = usePipelineDataContext()
   const { setStepDrawerOpen } = usePipelineViewContext()
 
-  // TODO: only 100 items
-  const { data: { body: pluginsResponse } = {} } = useListGlobalTemplatesQuery({
-    queryParams: { limit: 100, page: 1 }
+  const [page, setPage] = useState(1)
+  const [query, setQuery] = useState('')
+
+  const { data: { body: pluginsResponse, headers } = {}, isFetching } = useListGlobalTemplatesQuery({
+    queryParams: { limit: 100, page, query }
   })
 
+  const xNextPage = parseInt(headers?.get(PageResponseHeader.xNextPage) || '')
+  const xPrevPage = parseInt(headers?.get(PageResponseHeader.xPrevPage) || '')
+
+  const templatesSectionRef = useRef<HTMLDivElement | null>(null)
+
+  const harnessStepGroupsFiltered = useMemo(
+    () => harnessStepGroups.filter(harnessStepGroup => harnessStepGroup.identifier.includes(query)),
+    [query, harnessStepGroups]
+  )
+
+  const harnessStepsFiltered = useMemo(
+    () => harnessSteps.filter(harnessStep => harnessStep.identifier.includes(query)),
+    [query, harnessSteps]
+  )
+
   return (
     <StepsPalette.Root>
       <StepsPalette.Header>
         <StepsPalette.Title>Add Step</StepsPalette.Title>
-        {/* <Input placeholder="Search" />
-        <StepPaletteFilters /> */}
+        <Input
+          placeholder="Search"
+          onChange={value => {
+            setQuery(value.target.value)
```

Comment:
can we make use of `useCommonFilter` hook for this and page filter?

Reply:
for now, we are not using query params